docs/tutorial/flask.rst

```diff
@@ -0,0 +1,530 @@
+in your working directory with the ``.rock`` extension:
+
+.. literalinclude:: code/flask/task.yaml
+    :language: bash
+    :start-after: [docs:ls-rock]
+    :end-before: [docs:ls-rock-end]
+    :dedent: 2
+
+.. note::
+
+    If you changed the ``name`` or ``version`` in ``rockcraft.yaml`` or are
+    not on an ``amd64`` platform, the name of the ``.rock`` file will be
+    different for you.
+
+The rock needs to be copied to the MicroK8s registry so that it can be
+deployed in the Kubernetes cluster:
+
+.. literalinclude:: code/flask/task.yaml
+    :language: bash
+    :start-after: [docs:skopeo-copy]
+    :end-before: [docs:skopeo-copy-end]
+    :dedent: 2
+
+.. seealso::
+
+    See more: `skopeo <https://manpages.ubuntu.com/manpages/jammy/man1/skopeo.1.html>`_
+
+Create the charm
+================
+
+Create a new directory for the charm and go inside it:
+
+.. literalinclude:: code/flask/task.yaml
+    :language: bash
+    :start-after: [docs:create-charm-dir]
+    :end-before: [docs:create-charm-dir-end]
+    :dedent: 2
+
+We'll need a ``charmcraft.yaml``, ``requirements.txt`` and source code for
+the charm. The source code contains the logic required to operate the Flask
+application. Charmcraft will automate the creation of these files by using
+the ``flask-framework`` profile:
+
+.. literalinclude:: code/flask/task.yaml
+    :language: bash
+    :start-after: [docs:charm-init]
+    :end-before: [docs:charm-init-end]
+    :dedent: 2
+
+The files will automatically be created in your working directory.
+Pack the charm:
+
+.. literalinclude:: code/flask/task.yaml
+    :language: bash
+    :start-after: [docs:charm-pack]
+    :end-before: [docs:charm-pack-end]
+    :dedent: 2
+
+.. note::
+
+    Depending on your system and network, this step can take a couple
+    of minutes to finish.
+
+Once Charmcraft has finished packing the charm, you'll find a new file in your
+working directory with the ``.charm`` extension:
+
+.. literalinclude:: code/flask/task.yaml
+    :language: bash
+    :start-after: [docs:ls-charm]
+    :end-before: [docs:ls-charm-end]
+    :dedent: 2
+
+.. note::
+
+    If you changed the name in charmcraft.yaml or are not on the amd64 platform,
+    the name of the ``.charm`` file will be different for you.
+
+Deploy the Flask application
+============================
+
+A Juju model is needed to deploy the application. Let's create a new model:
+
+.. literalinclude:: code/flask/task.yaml
+    :language: bash
+    :start-after: [docs:add-juju-model]
+    :end-before: [docs:add-juju-model-end]
+    :dedent: 2
+
+.. warning::
+
+    If you are not on a host with the amd64 architecture, you will need to include
+    a constraint to the Juju model to specify your architecture. For example, for
+    the arm64 architecture, use
+    ``juju set-model-constraints -m flask-hello-world arch=arm64``.
+    Check the architecture of your system using ``dpkg --print-architecture``.
```

Comment:
Per our [style guide](https://docs.ubuntu.com/styleguide/en/#admonitions), this ought to be a different kind of admonition. A warning describes risk of damage to life, property, or data, but I don't see any negative consequences in this paragraph.

Since we don't have "information" as a type in RTD, "important" is probably the appropriate alternative: https://canonical-documentation-with-sphinx-and-readthedocscom.readthedocs-hosted.com/style-guide/#notes.

Reply:
Thanks for pointing this out! 

In the Django version of this tutorial, Daniele suggested that this note should be its own step (so that arm64 people don't miss the instruction and run into problems down the line). I would be fine with removing the warning (or important) dressing here

Reply:
Sounds good to me!